pkg/auth/handler/webapp/sso_callback.go

```diff
@@ -38,6 +41,22 @@ func (h *SSOCallbackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	s := session.GetSession(r.Context())
+
+	if state.AccountManagementToken != "" {
+		_, err = h.AccountManagement.FinishAddingIdentityOAuth(s, &accountmanagement.FinishAddingIdentityOAuthInput{
+			Token: state.AccountManagementToken,
+			Query: r.URL.Query().Encode(),
+		})
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
We should handle the error properly.

1. The error could be an API error. For example, linking to a OAuth account that is already associated with an existing Authgear user. This should be handled by showing the error in the original route where the add operation is triggered.
2. The error could be an OAuth error produced by the OAuth provider, such as complaining the client secret is invalid. Let's see how is this error handled in v1.
3. The error could be an I/O error. This error can be panic to let the panic middleware to handle.

Reply:
Updated to use authflow like error handling

*API error e.g. conflict identity*
![Screen Shot 2024-10-14 at 19 42 45](https://github.com/user-attachments/assets/137e27e0-d890-4800-86cc-4db8ebbd49c8)

*Unknown error e.g. secret error*
![Screen Shot 2024-10-14 at 19 42 38](https://github.com/user-attachments/assets/2f07557c-d265-429d-bcd9-704de999e797)